SB-36 : create assignment model and implement APIs

bandi-rajashree21 opened the pull request:
# Description

Create assignment model

Jira Issue [#SB-36](https://app.clickup.com/t/86cykwk5j)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Chore (not feature-related, CI, repo or versioning)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update


# Checklist:

- [x] I Have mentioned in the title of the PR the related Jira Issue
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

bandi-rajashree21 committed:
(feat) : create assignment model and updated schema

bandi-rajashree21 committed:
(feat) : create assignment model and updated schema

Manoo07 reviewed (commented):
Please make these changes!

bandi-rajashree21 committed:
(chore) : modified naming for fileUrl

bandi-rajashree21 committed:
(chore) : modified naming for fileUrl

BANDI-RAJASHREE commented:
> Please make these changes!

bandi-rajashree21 committed:
(chore): rename submissionFileUrl with proper naming

bandi-rajashree21 committed:
(chore): rename submissionFileUrl with proper naming

bandi-rajashree21 committed:
(feat) : add assignment APIs

bandi-rajashree21 committed:
(chore) : improve assignments api's

Manoo07 committed:
Merge branch 'main' into SB-36

Manoo07 committed:
Threads (#61)

* (feat) : add thread like, delete endpoints and also DB change for classroom specific threads

* (feat) : Threads

* feat: enhance thread filtering with global vs classroom-specific logic

- Implement clear separation between global threads and classroom-specific threads
- Add global threads filter (classroomId = null, parentId = null)
- Improve classroom-specific thread filtering
- Add comprehensive logging for filter operations
- Enhance unit filtering logic for better thread organization

* feat: remove auth middleware from thread deletion route

- Remove authentication requirement for thread deletion endpoint
- Simplify thread deletion access control
- Update route security configuration

* (feat) : update .env.example

* (feat) : add filter to threads

* Update DAO files and thread controller with latest changes

* (fix) : Threads seperation from global & classroom based threads

* (feat) : add Prisma Singleton

---------

Co-authored-by: mboinapally <mboinapally@osidigital.com>

manohar-mano committed:
(refactor) : add Singleton object for AuthController

bandi-rajashree21 committed:
(fix) : remove duplicate imports (#62)

* (fix) : remove duplicate imports

* (fix) : remove duplicate imports

Manoo07 committed:
(fix) : reverted to stable versions for packages

bandi-rajashree21 committed:
Sb 108 (#63)

* (chore) : add enrollmentNo to eligible Students

* (fix) : likes for thread and thread Comments

bandi-rajashree21 committed:
(style) : format entire codebase (#64)

bandi-rajashree21 committed:
(fix) : typo in virtualClassroomDAO (#65)

bandi-rajashree21 committed:
(chore) : resolve conflicts

bandi-rajashree21 committed:
(chore) : resolve merge conflicts
